metadata/src/main/java/org/apache/kafka/metadata/KafkaConfigSchema.java

```diff
@@ -166,9 +166,11 @@ public Map<String, ConfigEntry> resolveEffectiveTopicConfigs(
         ConfigDef configDef = configDefs.getOrDefault(ConfigResource.Type.TOPIC, EMPTY_CONFIG_DEF);
         HashMap<String, ConfigEntry> effectiveConfigs = new HashMap<>();
         for (ConfigDef.ConfigKey configKey : configDef.configKeys().values()) {
-            ConfigEntry entry = resolveEffectiveTopicConfig(configKey, staticNodeConfig,
-                dynamicClusterConfigs, dynamicNodeConfigs, dynamicTopicConfigs);
-            effectiveConfigs.put(entry.name(), entry);
+            if (!configKey.internalConfig) {
```

Comment:
I believe the CreateTopic result in KRaft does not respect not including internal configs. Let me know if this is untrue or if you think it ought to be added someplace else 😊

Reply:
Could I know why we should do this change? For broken tests? I'm not 100% sure if this change changes some behavior or not. If so, I'd suggest we make the config as non-internal config. We plan to make it complete in v3.9.0 anyway. Thoughts?

Reply:
Yup, this was breaking a test in one of the previous test runs on the PR. Happy to go with not defining the configuration as internal 😊